### PR TITLE
fix self_tests

### PR DESCRIPTION
**Pull request by rchaser53**
fix: #4017

### CODE REVIEW

File: rustfmt-core/rustfmt-bin/src/bin/main.rs [hunk 1]
```diff
@@ -92,7 +92,6 @@ struct Opt {
     verbose: bool,
 
     // Nightly-only options.
-
```
Comment (calebcartwright) [inline rustfmt-core/rustfmt-bin/src/bin/main.rs]: Might be nice to keep this (and the `Positional args` on line 125/126 below) separate so they don't seem to be pinned to a specific opt?

File: rustfmt-core/rustfmt-lib/src/test/mod.rs [hunk 2]
```diff
@@ -385,7 +389,8 @@ fn self_tests() {
     if !is_nightly_channel!() {
         return;
     }
-    let mut files = get_test_files(Path::new("tests"), false);
+    let skip_file_white_list = ["target", "tests"];
+    let mut files = get_test_files(Path::new("../../rustfmt-core"), true, &skip_file_white_list);
     files.push(PathBuf::from("src/lib.rs"));
```
Comment (calebcartwright) [inline rustfmt-core/rustfmt-lib/src/test/mod.rs]: So I was also originally thinking that we could restore `self_tests` by conditionally enabling the `recursive` Config setting that could be controlled via additional flags to `check_files()` and then to `idempotent_check()` vs. running in non-recursive mode and collecting each individual file. 

However, if we are going to use the updated form of `get_test_files` to collect every file, then I don't think there's a need to have this line to push the entry `lib.rs` file again (it should already be there), right?

File: rustfmt-core/rustfmt-bin/src/bin/main.rs (hunk 1 above)
Comment (rchaser53) [reply rustfmt-core/rustfmt-bin/src/bin/main.rs]: Now rustfmt removes the newline in this case.
So if I insert newline here, self_tests is failed...
I think this is the rustfmt bug.
maybe related: #4012

File: rustfmt-core/rustfmt-lib/src/test/mod.rs (hunk 2 above)
Comment (rchaser53) [reply rustfmt-core/rustfmt-lib/src/test/mod.rs]: Certainly. I fixed it. Thank you for the review.

**Pull request merged**